packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -6244,13 +6249,26 @@ class WidgetToRenderBoxAdapter extends LeafRenderObjectWidget {
   /// tree will be dirty.
   final VoidCallback? onBuild;
 
+  /// Called when it is safe to dispose of the render object and its
+  /// descendants.
+  ///
+  /// Typically, this should dispose of any children attached to the
```

Comment:
The two paragraphs sound a little contradictory: One the one hand its safe to dispose the render object (first paragraph) on the other hand it must not be disposed (second paragraph)?

Reply:
You're right, I've updated this to try to clarify it and added links to some relevant areas.